File: sorting_algorithm_python/insertion.py
arr=[2,4,5,6,3,1,7,8,9,0]

def insertion(arr):
    for i in range(1,len(arr)):
        sorted=arr[i]
        j=i-1
        while(j>=0 and arr[j]>sorted):
            arr[j+1]=arr[j]
            j=j-1
        arr[j+1]=sorted

    print(arr)

insertion(arr)
            

File: sorting_algorithm_c/insertion.c
#include<stdio.h>
#define size 10


void insertion(int arr[size],int n){
    int i,j;
    int temp;
    for(i=1;i<=n-1;i++){
        temp=arr[i];
        j=i-1;
        while(j>=0&&arr[j]>temp){
            arr[j+1]=arr[j];
            j--;
        }
        arr[j+1]=temp;
        
        
    }

}

void print(int arr[size],int n){
    int i;
    printf("The sorted array is: ");
    for(i=0;i<n;i++){
        printf("%d ",arr[i]);
        

    }
}

int main(){
    int arr[size]={2,4,3,1,5,7,6,9,8,0};

    insertion(arr,size);
    print(arr,size);


}